src/command/SimpleCommandMap.php

```diff
@@ -243,7 +244,10 @@ public function dispatch(CommandSender $sender, string $commandLine) : bool{
 		try{
 			$target->execute($sender, $sentCommandLabel, $args);
 		}catch(InvalidCommandSyntaxException $e){
-			$sender->sendMessage($this->server->getLanguage()->translateString("commands.generic.usage", [$target->getUsage()]));
+			if($sender instanceof Player) {
```

Comment:
This check is a bit sloppy in my opinion and can be avoided by moving the `getLanguage()` function to the CommandSender class

Reply:
I agree that the design looks terrible , but what about the usual sendMessage that is used in Player.php? Not applicable, if you move the $language variable to CommandSender, you can use this variable from Player, but it is not logically related that the player's language is in CommandSender

Reply:
If you go to extreme measures, you can pass the Language when initializing CommandSender, so that it is stored there in public access

Reply:
`CommandSender` is an interface not a class and thus "move the $language variable to CommandSender" does not make sense.

The goal of moving `getLanguage()` to `CommandSender` is so that, by Liskov Substitution Principle, one can send a localized message to a command sender regardless whether the command sender is a Player. For example, if the command was sent by some `CommandSender` implementation not defined by pocketmine, it is still possible to let them have their own language.

Always keep in mind that `CommandSender` is not a sealed interface, and any plugin may add their implementations.

Reply:
> `CommandSender` is an interface not a class and thus "move the $language variable to CommandSender" does not make sense.
> 
> The goal of moving `getLanguage()` to `CommandSender` is so that, by Liskov Substitution Principle, one can send a localized message to a command sender regardless whether the command sender is a Player. For example, if the command was sent by some `CommandSender` implementation not defined by pocketmine, it is still possible to let them have their own language.
> 
> Always keep in mind that `CommandSender` is not a sealed interface, and any plugin may add their implementations.

In CommandSender, you can make the getLanguage () method, which is already available in Player and Server

Reply:
What if I want to localize the language for the console command sender? Such as a situation where a command is sent via console?

Reply:
In that case, why even require a `Language` object? Why not just expose a method that takes translation key + arguments and returns the translated value, without exposing the actual language type?

Reply:
There are cases where it's necessary to pre-translate, due to the translation system not supporting nested translations, such as [this](https://github.com/pmmp/pocketmine-mp/blob/3473254d01e5012a39bf9f714a6c87d58907c4ab/src/command/Command.php#L232).